subsys/net/ip/net_pkt.c

```diff
@@ -974,6 +974,13 @@ static size_t pkt_buffer_length(struct net_pkt *pkt,
 		if (net_if_l2(net_pkt_iface(pkt)) ==
 		    &NET_L2_GET_NAME(ETHERNET)) {
 			max_len += sizeof(struct net_eth_hdr);
+#if IS_ENABLED(CONFIG_NET_DSA) && IS_ENABLED(CONFIG_DSA_KSZ8794)
+			/*
+			 * Extend the max pkt length for DSA (KSZ8794) by one
```

Comment:
if such tag is found in all DSA, no need to tell about a specific one.

If not, then that is not a proper solution (the tag would need no stay always managed and hidden in drivers/ethernet/dsa)

Reply:
I've added following solution to the `dsa.h` file.

`#if defined(CONFIG_DSA_KSZ8794) && defined(CONFIG_DSA_KSZ8794_TAIL_TAGGING)`
`#define _DSA_TAG_SIZE 1`
`#else`
`#define _DSA_TAG_SIZE 0`
`#endif`

And then only use `_DSA_TAG_SIZE` in generic code.

Reply:
The above change has been applied to updated PR.